server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java

```diff
@@ -150,52 +153,47 @@ private String removeUser(Request request, Response response) {
                 .message("The user " + username + " does not exists")
                 .cause(e)
                 .haltError();
-        } catch (UserService.InvalidUsername e) {
-            LOGGER.info("Invalid username", e);
-            throw ErrorResponder.builder()
-                .statusCode(HttpStatus.BAD_REQUEST_400)
-                .type(ErrorType.INVALID_ARGUMENT)
-                .message("Invalid username: it should be between 1 and 255 long without '/'")
-                .cause(e)
-                .haltError();
         } catch (IllegalArgumentException e) {
             LOGGER.info("Invalid user path", e);
             throw ErrorResponder.builder()
                 .statusCode(HttpStatus.BAD_REQUEST_400)
                 .type(ErrorType.INVALID_ARGUMENT)
-                .message("Invalid user path")
+                .message(e.getMessage())
                 .cause(e)
```

Comment:
Won't this lead to a duplicated message & cause?

Won't it be better to rely on default IllegalArgumentException handling here? (remove this catch block)

Reply:
the default `IllegalArgumentException` handling only return a very comm message `Invalid arguments supplied in the user request`. Which is not really meaningful to me, so I prefer to catch.

Reply:
```
message: Invalid arguments supplied in the user request
cause: Username can not contain `@` when virtualhosting is disabled
```

Hint: a generic message is OK when we get the right cause!